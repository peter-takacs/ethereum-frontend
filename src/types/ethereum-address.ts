export class Address {
    private readonly address: string;

    public static isValid(address: string) {
        return /^0x[a-fA-F0-9]{40}$/.test(address);
    }

    public static fromBytes(bytes: Uint8Array): Address {
        if (bytes.length !== 20) {
            throw new Error(`Invalid amount of bytes supplied to ethereum address.`)
        }
        let address = '0x';
        for (const byte of bytes) {
            address = address + byte.toString(16);
        }
        return new Address(address);
    }

    constructor (address: string) {
        if (!Address.isValid(address)) {
            throw new Error(`Invalid address "${address}" supplied to ethereum address constructor.`)
        }
        this.address = address;
    }

    public toString() {
        return this.address;
    }

    public equals(other: Address | null) {
        if (other == null) {
            return false;
        }
        return this.address === other.address;
    }
}