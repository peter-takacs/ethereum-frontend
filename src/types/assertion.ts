import { Address } from "./ethereum-address";

export interface Assertion {
    certificate: string;
    issuer: Address;
}