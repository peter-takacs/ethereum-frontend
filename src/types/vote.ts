import { Address } from "./ethereum-address";

export enum VoteStatus {
    Pending = 0,
    Accept = 1,
    Reject = 2
}

export interface CandidateVotes {
    candidate: Address;
    votes: Map<Address, VoteStatus>;
}