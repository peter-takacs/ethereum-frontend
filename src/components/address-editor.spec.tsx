import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Address } from '../types/ethereum-address';
import AddressEditor from './address-editor';

test('It renders correctly with an empty address', () => {
    const tree = renderer.create(
        <AddressEditor address={null} onChange={() => {}}/>
    )
    expect(tree.toJSON()).toMatchSnapshot();
})


test('It renders correctly with a correct address', () => {
    const address = new Address('0x0000111122223333444455556666777788889999');
    const tree = renderer.create(
        <AddressEditor address={address} onChange={() => {}}/>
    )
    expect(tree.toJSON()).toMatchSnapshot();
})