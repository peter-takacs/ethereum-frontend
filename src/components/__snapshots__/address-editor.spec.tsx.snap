// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly with a correct address 1`] = `
<div>
  <div
    className="MuiFormControl-root-1"
  >
    <div
      className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-underline-9"
    >
      <input
        aria-invalid={false}
        className="MuiInput-input-13"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="0x0000111122223333444455556666777788889999"
      />
    </div>
  </div>
</div>
`;

exports[`It renders correctly with an empty address 1`] = `
<div>
  <div
    className="MuiFormControl-root-1"
  >
    <div
      className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-underline-9"
    >
      <input
        aria-invalid={false}
        className="MuiInput-input-13"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
